From b80115dc6e368e88629d0ced75cd5cae28c6d9bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 13 Mar 2020 15:37:42 +0100
Subject: [PATCH] dateMenu: Don't ellipsize world clock time/tz

If we need to ellipsize, it should be the location name, not the time
or timezone offset.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1090
---
 js/ui/dateMenu.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/js/ui/dateMenu.js b/js/ui/dateMenu.js
index c6c1b5bb5a..cdbe0ac415 100644
--- a/js/ui/dateMenu.js
+++ b/js/ui/dateMenu.js
@@ -189,6 +189,9 @@ class WorldClocksSection extends St.Button {
                                     x_align: Clutter.ActorAlign.END,
                                     y_align: Clutter.ActorAlign.CENTER });
 
+            time.clutter_text.ellipsize = Pango.EllipsizeMode.NONE;
+            tz.clutter_text.ellipsize = Pango.EllipsizeMode.NONE;
+
             if (this._grid.text_direction == Clutter.TextDirection.RTL) {
                 layout.attach(tz, 0, i + 1, 1, 1);
                 layout.attach(time, 1, i + 1, 1, 1);
-- 
2.24.1

